actionmailer/lib/action_mailer/message_delivery.rb

```diff
@@ -14,30 +13,35 @@ module ActionMailer
   #   Notifier.welcome(User.first).deliver_later # enqueue email delivery as a job through Active Job
   #   Notifier.welcome(User.first).message       # a Mail::Message object
   class MessageDelivery < Delegator
-    def initialize(mailer, mail_method, *args) #:nodoc:
-      @mailer = mailer
-      @mail_method = mail_method
-      @args = args
-      @obj = nil
+    def initialize(mailer_class, action, *args) #:nodoc:
+      @mailer_class, @action, @args = mailer_class, action, args
+
+      # The mail is only processed if we try to call any methods on it.
+      # Typical usage will leave it unloaded and call deliver_later.
+      @processed_mailer = nil
+      @mail_message = nil
     end
 
+    # Method calls are delegated to the Mail::Message that's ready to deliver.
     def __getobj__ #:nodoc:
-      @obj ||= begin
-                 mailer = @mailer.new
-                 mailer.process @mail_method, *@args
-                 mailer.message
-               end
+      @mail_message ||= processed_mailer.message
```

Comment:
Separate out mail processing so we can dispatch exception handling to the mailer instance.